packages/eas-cli/src/credentials/ios/actions/new/SetupAdhocProvisioningProfile.ts

```diff
@@ -1,77 +1,69 @@
 import Log from '../../../../log';
 import { confirmAsync } from '../../../../prompts';
 import { Context } from '../../../context';
 import { AppLookupParams } from '../../api/GraphqlClient';
-import { AppleProvisioningProfileMutationResult } from '../../api/graphql/mutations/AppleProvisioningProfileMutation';
-import { AppleDeviceFragmentWithAppleTeam } from '../../api/graphql/queries/AppleDeviceQuery';
-import { AppleProvisioningProfileQueryResult } from '../../api/graphql/queries/AppleProvisioningProfileQuery';
-import { ProvisioningProfileStoreInfo } from '../../appstore/Credentials.types';
-import { ProfileClass } from '../../appstore/provisioningProfile';
-import { AppleTeamMissingError, MissingCredentialsNonInteractiveError } from '../../errors';
+import { MissingCredentialsNonInteractiveError } from '../../errors';
+import { validateProvisioningProfileAsync } from '../../validators/validateProvisioningProfile';
 import { resolveAppleTeamIfAuthenticatedAsync } from './AppleTeamUtils';
-import { chooseDevices } from './DeviceUtils';
-import { doUDIDsMatch, isDevPortalAdhocProfileValid } from './ProvisioningProfileUtils';
+import { assignBuildCredentialsAsync, getBuildCredentialsAsync } from './BuildCredentialsUtils';
+import { chooseDevices, formatDeviceLabel } from './DeviceUtils';
 import { SetupDistributionCertificate } from './SetupDistributionCertificate';
 
 export class SetupAdhocProvisioningProfile {
   constructor(private app: AppLookupParams) {}
 
   async runAsync(ctx: Context): Promise<IosAppBuildCredentialsFragment> {
+    const areBuildCredentialsSetup = await this.areBuildCredentialsSetupAsync(ctx);
+
     if (ctx.nonInteractive) {
-      try {
-        return await this.runNonInteractiveAsync(ctx);
-      } catch (err) {
-        if (err instanceof AppleTeamMissingError) {
-          throw new MissingCredentialsNonInteractiveError();
-        }
-        throw err;
+      if (areBuildCredentialsSetup) {
+        return nullthrows(await getBuildCredentialsAsync(ctx, this.app, IosDistributionType.AdHoc));
+      } else {
+        throw new MissingCredentialsNonInteractiveError(
+          'Provisioning profile is not configured correctly. Please run this command again in interactive mode.'
+        );
       }
-    } else {
-      return await this.runInteractiveAsync(ctx);
     }
-  }
-
-  private async runNonInteractiveAsync(ctx: Context): Promise<IosAppBuildCredentialsFragment> {
-    // 1. Setup Distribution Certificate
-    await new SetupDistributionCertificate(this.app, IosDistributionType.AdHoc).runAsync(ctx);
 
-    // 2. Fetch profile from EAS servers
-    const currentProfile = await ctx.newIos.getProvisioningProfileAsync(
+    const currentBuildCredentials = await getBuildCredentialsAsync(
+      ctx,
       this.app,
       IosDistributionType.AdHoc
     );
-
-    if (!currentProfile) {
-      throw new MissingCredentialsNonInteractiveError();
+    if (areBuildCredentialsSetup) {
```

Comment:
This will only check the provisioning profile and because `SetupDistributionCertificate` was not called before we don't know if it's valid. I know that profile validation is using dist certs so they will be partially validated there, but I'm not sure if  we handle all cases there